### DISCUSSION TITLE
[Feature Request] Add delay/check for checking a monitors availability

### DISCUSSION THREAD

**Discussion by SouljaVR** [LizardByte/community#604] [Sunshine]
[Feature Request]

Hi - I made two simple scripts, one that disables all my physical monitors and enables my virtual monitor, and then the other script does vice versa. I want to have these scripts automatically execute based on connection/disconnection to sunshine. Doing this is no issue of course, however the main problem is Sunshine immediately attempts to connect to the display when initializing the stream which causes one of two things to happen:

1. If I specify my virtual display in the Adapter Name tab, it does not find it as the process of enabling the virtual monitor/disabling the physical monitors takes a few seconds. But since Sunshine does not wait to allow for the monitor to be available it doesn't ever see it, so the connection fails.

2. If I leave the Adapter Name empty, it ends up connecting to my main monitor instead which stops the script from working. It cannot disable my primary monitor if Sunshine is connected to it, so this still doesn't work ideally at all.

A simple solution would be adding some kind of delay system so we can specify how long Sunshine should wait AFTER executing the commands to attempt to connect to the display. Or perhaps we could have a retry system that allows for the Adapter Name input to be retried X number of times over X period before it truly counts it as a failure.

The ideal outcome is that when my device connects to Sunshine, sunshine executes my bat script to disable my physical monitors and enable my virtual monitor. This process takes about 5 seconds. Then once this is done Sunshine should then be able to see and use the virtual monitor fine. So essentially, we want a way to delay how quickly Sunshine tries to connect/capture the display so the system can be ready first. This feature would be very helpful, thank you.

If such a feature already exists, please help me out as right now I haven't found a good solution. The best solution as of now is to just always keep the virtual monitor enabled and at a low resolution/refresh rate but this causes a lot of annoyances when I am trying to use my PC. Of course, I could just manually run the script beforehand too which is what I am doing at the moment, but this isn't very streamlined. Thank you and hopefully you will consider adding this. Also thank you for the amazing work with Sunshine overall, it has been really cool being able to play my PC games with such solid performance through my phone, away from my PC :)

**Comment by ReenigneArcher** [comment]
This is on our roadmap: https://github.com/LizardByte/roadmap/issues/38

I think the proposed solution would solve your problem, but not as you imagined. Instead there will be different points during the launch process where commands could be run.
